rubix-bookkeeper/src/test/java/com/qubole/rubix/bookkeeper/TestBookKeeper.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2016. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.bookkeeper;
+
+import com.codahale.metrics.MetricRegistry;
+import com.qubole.rubix.core.ClusterManagerInitilizationException;
+import com.qubole.rubix.core.utils.DeleteFileVisitor;
+import com.qubole.rubix.core.utils.DummyClusterManager;
+import com.qubole.rubix.spi.CacheConfig;
+import com.qubole.rubix.spi.ClusterManager;
+import com.qubole.rubix.spi.ClusterType;
+import org.apache.hadoop.conf.Configuration;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Created by Abhishek on 6/15/18.
+ */
+public class TestBookKeeper
```

Comment:
This is a good start. 

Can you test Presto & H2 cluster types as well  ? I understand that you cant use the default classes for these types.
One way to get around that is to:

    public void testGetClusterManagerInValidInstance() throws Exception
    {
        ClusterType type = ClusterType.PRESTO_CLUSTER_MANAGER; //or H2
        CacheConfig.setDummyClusterManager(conf, "com.qubole.rubix.core.test.PrestoDummyClusterManager");
        BookKeeper bookKeeper = new CoordinatorBookKeeper(conf, metrics);

        ClusterManager manager = bookKeeper.getClusterManagerInstance(type, conf);
        //assert that the class is of the right type. 
    }

Reply:
Done